### PR TITLE
feat: siteFile favicon config support function type

### PR DESCRIPTION
**Pull request by kuitos**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

We could use a function to customize favicon config

### CODE REVIEW

File: app/middleware/site_file.js [hunk 1]
```diff
@@ -14,6 +14,10 @@ module.exports = options => {
     // '/favicon.ico': 'https://eggjs.org/favicon.ico',
     // content is url
     if (typeof content === 'string') return ctx.redirect(content);
+    if (typeof content === 'function') {
+      const faviconUrl = content(ctx);
```
Comment (fengmk2) [inline app/middleware/site_file.js:18]: faviconUrl 会不会返回空不进行跳转？

Comment (kuitos) [reply app/middleware/site_file.js]: 这里我改了一下，复用了一下之前的 string 检查逻辑

**Review by fengmk2 — APPROVED** [review]

File: app/middleware/site_file.js [hunk 2]
```diff
@@ -8,10 +8,12 @@ module.exports = options => {
...
     if (ctx.path[0] !== '/') return next();
 
-    const content = options[ctx.path];
+    let content = options[ctx.path];
     if (!content) return next();
 
-    // '/favicon.ico': 'https://eggjs.org/favicon.ico',
+    // '/favicon.ico': 'https://eggjs.org/favicon.ico' or '/favicon.ico': (ctx) => 'https://eggjs.org/favicon.ico'
+    // content is function
+    if (typeof content === 'function') content = content(ctx);
```
Comment (fengmk2) [inline app/middleware/site_file.js:16]: 清晰多了。如果后续 content 需要异步读取数据，再来一个 pr 改进。

Comment (kuitos) [reply app/middleware/site_file.js:16]: 你提醒我了，现在的需求其实就需要异步的。我现在就改一下😂

Comment (fengmk2) [reply app/middleware/site_file.js]: > 你提醒我了，现在的需求其实就需要异步的。我现在就改一下😂

😄

**Review by fengmk2 — APPROVED** [review]

**Pull request merged**